substratevm/src/com.oracle.svm.reflect/src/com/oracle/svm/reflect/serialize/hosted/SerializationFeature.java

```diff
@@ -205,10 +219,100 @@ private void abortIfSealed() {
         UserError.guarantee(!sealed, "Too late to add classes for serialization. Registration must happen in a Feature before the analysis has finished.");
     }
 
+    @Override
+    public void registerAllAssociatedClasses(ConfigurationCondition condition, Class<?> clazz) {
+        String targetClassName = clazz.getName();
+        // If the serialization target is primitive, it needs to get boxed, because the target is
+        // always an Object.
+        if (clazz.isPrimitive()) {
+            Class<?> boxedType = null;
+            switch (targetClassName) {
```

Comment:
Please use something like https://github.com/oracle/graal/blob/d08f7ad5797ac68431d8fc92b8ca3d4f6caf724b/truffle/src/com.oracle.truffle.host/src/com/oracle/truffle/host/HostExecuteNode.java#L484..L505

Reply:
I found `jdk.vm.ci.meta.JavaKind` has proper methods to do the work:
https://github.com/graalvm/labs-openjdk-11/blob/master/src/jdk.internal.vm.ci/share/classes/jdk.vm.ci.meta/src/jdk/vm/ci/meta/JavaKind.java#L253-L274
and
https://github.com/graalvm/labs-openjdk-11/blob/master/src/jdk.internal.vm.ci/share/classes/jdk.vm.ci.meta/src/jdk/vm/ci/meta/JavaKind.java#L291-L293

Reply:
Yes that's even better.